### DISCUSSION TITLE
child state become nil after navigate to newView

### DISCUSSION THREAD

**Discussion by saroar** [pointfreeco/swift-composable-architecture#791] [Q&A]
I am using his NavigationExtension @darrarski
https://github.com/darrarski/darrarski/tree/main/2021/04/Thoughts-on-SwiftUI-navigation
here is my app test https://github.com/AddaMeSPB/AddaMeIOS/blob/feature/git_workflow/AddameSPM/Tests/EventViewTests/EventsViewTests.swift
my child state is become nil imminently 

https://user-images.githubusercontent.com/8770772/133213979-6dc3e1af-4b3e-485c-af5d-b3cd0202772c.MP4

thanks advance for help

**Comment by darrarski** [comment]
Did you try to debug your reducer so you'll know what actions are sent to the store, and how the state is mutated? This may help finding the issue.

When it comes to my "Thought on SwiftUI navigation" article, I've made a Swift Package you can use in your app: https://github.com/darrarski/swift-composable-presentation.

**Reply by saroar** [reply]
@darrarski yes I am using your extension on my project thanks for nice extension
yes I debug it but can't find any issue still now

**Reply by darrarski** [reply]
Could you add `.debug()` to your reducer and attach the console log that shows what's going on when you navigate?

**Reply by saroar** [reply]
https://gist.github.com/saroar/594f731fe89c8a3a22b439c74be9dcef#file-eventreducer-swift-L135


```
.navigate(
      using: store.scope(
        state: \.eventFormState,
        action: EventsAction.eventForm
      ),
      destination: EventFormView.init(store:),
      onDismiss: {
        ViewStore(store.stateless).send(.eventFormView(isNavigate: false)) -> from EventView when I commend this line 
      }
    )
```
still back to root view and logs nothing especial

**Reply by saroar** [reply]
@darrarski awesome package like it:) I already it in my app 
now after debug lots I found call any action child view my child view back to root view :( very strange behave

**Reply by saroar** [reply]
https://user-images.githubusercontent.com/8770772/133293229-72b8d3a0-6734-47a4-b0a3-4b4ed89afb39.mp4

I am trying more debug rit now

**Reply by saroar** [reply]
before move to my EventView my eventState become nil its very strange

**Comment by mbrandonw** [comment]
Have you tried applying `StackNavigationViewStyle` to your base navigation view? This is what we do in isowords:

https://github.com/pointfreeco/isowords/blob/dfaf3d3d36cce1b47eb032d4f13e4d737c185c81/Sources/AppFeature/AppView.swift#L443-L446

This can fix a variety of weird navigation issues.

**Reply by saroar** [reply]
from morning I am trying to figure out problem found some problem but don't know reason
thanks for reply @mbrandonw yes I follow isowords code some times 
I already have it here is it please take look if u found something wrong 

https://github.com/AddaMeSPB/AddaMeIOS/blob/feature/git_workflow/AddameSPM/Sources/TabsView/TabsView.swift#L96

**Reply by saroar** [reply]
```
extension EventsAction {
  static func view(_ localAction: EventView.ViewAction) -> Self {
    switch localAction {
    case .alertDismissed:
      return .alertDismissed
    case .dismissEventDetails:
      return .dismissEventDetails
    ..................
    }
}
``` is it normal ?

**Reply by saroar** [reply]
here is what happing when I click any action on my child view 
https://gist.github.com/saroar/8bebb7363a56c15ff6e172a44d773b8f

**Reply by saroar** [reply]
then main reason this bug is called child view any action call parent view onDismiss action which is call 

```
onDismiss: {
        ViewStore(store.stateless).send(.eventFormView(isNavigate: false)) -> from EventView when I commend this line 
      }
```
